pkg/utils/utils.go

```diff
@@ -51,7 +51,12 @@ func GenerateLabel(labels *map[string]string, name string, value string) {
 }
 
 // GenerateRuntimeObjectName generates name using BackstageCR name and objectType which is ConfigObject Key without '.yaml' (like 'deployment')
-func GenerateRuntimeObjectName(backstageCRName string, objectType string) string {
+func GenerateRuntimeObjectName(backstageCRName string, objectType string, crNameAsPrefix bool) string {
```

Comment:
I would really like to have it unified, as discussed and not to make us remember what is the pattern for each object type.  I do not think it impacts on functionality, the only drawback is that we leave old objects which can be solved by:
- upgrade documentation (recommending to remove it manually when upgrade from 1.1)
- additional ckeck/remove  (but I'd avoid it).

Reply:
> I would really like to have it unified, as discussed and not to make us remember what is the pattern for each object type. I do not think it impacts on functionality
>
Well, in some cases, it impacted functionally. For example, I had to revert the DB Secret name to make sure a new Secret was not being created. Otherwise, connection to the existing DB would just fail.
I agree with you (at least in the medium/long run). But at the same time, found it really confusing to have both old and new objects in the cluster (without anything conveying which ones are deprecated).
I remember from a discussion someone edited the `<cr_name>-dynamic-plugins` ConfigMap (name in 1.1) to update their list of dynamic plugins. How would we handle this now that the name has changed to `backstage-dynamic-plugins-<cr_name>` in 1.2?

This is not ideal, but for the upcoming 1.2, I think it is safer to keep the existing naming to minimize the risk of additional bugs.

![image](https://github.com/janus-idp/operator/assets/593208/0cbad519-e839-4e7b-818b-750ceb648b03)

> the only drawback is that we leave old objects which can be solved by:
> 
> * upgrade documentation (recommending to remove it manually when upgrade from 1.1)
> * additional ckeck/remove  (but I'd avoid it).

The problem I foresee is with the Automatic Upgrade strategy with OLM, which, in my understanding, is much appreciated by users. IMO, we cannot claim backward compatibility and expect CR users to perform manual cleanup operations after an upgrade of the application (potentially automatic).
Or maybe a K8s Job (created by the operator) that does this migration automatically might make sense.
But that would be a separate issue.

Reply:
To make it clear

What is important to me is that the names of objects generated by Operator have unified pattern.
I.e. (as discussed) backstage-* or backstage-psql-* shape.
It is important for me that all of auto-generated items looks really as auto-generated, it also will simplify support.
And I think the sooner we do it the better (i.e. if we do not do it now it is a big chance it will live forever).

For your notes:

I think DB secret should not be affected, as I can see 1.1 name is secret/backstage-psql-secret-bs1 which fits the pattern, no?

For dynamic-plugins, I am not sure I understand the problem.
If you mean user change operator-config's configmap or secret - I do not think it will work anyway since it will be changed to desired state (from  operator-config). Or you meant something else? 

So far I can not see any backward compatibility problem.
I do not think existence of "old" objects can be taken as incompatible thing. 
Yes, we need to mention it in the Release Notes and recommend to clean it (I would rather not spend time for dedicated tooling for it).

Reply:
What pushes me to stick to the previous naming in 1.1 is because this is the second issue caused by this naming change (first one was https://github.com/janus-idp/operator/issues/355). And I wanted to prevent further issues due to this.
But alright, let's go ahead with your suggestions. I'll update the release notes text on the JIRA issue accordingly.
Thanks.